src/backend/common/helpers/matchstats_helper.py

```diff
@@ -8,29 +8,146 @@
+        lambda match, color: match.score_breakdown[color].get(comp, 0)
+    )
+
+    EVERGREEN: Dict[StatType, StatAccessor] = {
+        StatType.OPR: OPR_ACCESSOR,
+        StatType.DPR: DPR_ACCESSOR,
+        StatType.CCWM: CCWM_ACCESSOR,
+    }
+
+    COMPONENTS = {
+        2020: {
+            "Total Power Cell Points": lambda match, color: (
+                match.score_breakdown[color].get("autoCellPoints")
+                + match.score_breakdown[color].get("teleopCellPoints")
+            ),
+        },
+        2019: {
+            "Cargo + Panel Points": lambda match, color: (
+                match.score_breakdown[color].get("cargoPoints")
+                + match.score_breakdown[color].get("hatchPanelPoints")
+            )
+        },
+    }
+
+    @classmethod
+    def calculate_matchstats(
+        cls, matches: List[Match], skip_coprs: bool = False, keyed: bool = True
+    ) -> EventMatchstats:
+        return EventMatchstats(
+            oprs=cls.calculate_oprs(matches, keyed=keyed),
+            dprs=cls.calculate_dprs(matches, keyed=keyed),
+            ccwms=cls.calculate_ccwms(matches, keyed=keyed),
+            coprs={} if skip_coprs else cls.calculate_coprs(matches, keyed=keyed),
+        )
+
+    @classmethod
+    def calculate_oprs(cls, matches: List[Match], keyed: bool = True) -> TeamStatMap:
+        return cls.calculate_stat(matches, cls.OPR_ACCESSOR, keyed=keyed)
+
     @classmethod
-    def build_team_mapping(
+    def calculate_dprs(cls, matches: List[Match], keyed: bool = True) -> TeamStatMap:
+        return cls.calculate_stat(matches, cls.DPR_ACCESSOR, keyed=keyed)
+
+    @classmethod
+    def calculate_ccwms(cls, matches: List[Match], keyed: bool = True) -> TeamStatMap:
+        return cls.calculate_stat(matches, cls.CCWM_ACCESSOR, keyed=keyed)
+
+    @classmethod
+    def calculate_coprs(
+        cls, matches: List[Match], keyed: bool = True
+    ) -> Dict[Component, TeamStatMap]:
+        coprs = OrderedDict()
+
+        # If there is not valid data for COPRs, skip
+        if not (
+            matches is not None
+            and len(matches) > 0
+            and matches[0].score_breakdown is not None
+        ):
+            return coprs
+
+        # Specific components specified in cls
+        year = matches[0].year
+        if year in cls.COMPONENTS.keys():
+            for title, accessor in cls.COMPONENTS[year].items():
+                coprs[title] = cls.calculate_stat(matches, accessor, keyed=keyed)
+
+        # for each string-like key in the score_breakdown object
+        # (just take the red score_breakdown from match 0, it's an arbitrary selection)
+        for component, value in none_throws(matches[0].score_breakdown)[
+            AllianceColor.RED
+        ].items():
+            # if the component is an int or a float, we can compute it easily
+            # (as opposed to, say, a string)
+            if isinstance(value, int) or isinstance(value, float):
```

Comment:
One thought here is we could attempt to cast to a float, and if our cast succeeds, we're good to go. Otherwise we can't count it. That way we can support more types (like strings) without needing to add explicit support.